src/edgekv/ekv-cli-main.ts

```diff
@@ -0,0 +1,252 @@
+  .option('--edgerc <path>', 'Use edgerc file for authentication.')
+  .option('--section <name>', 'Use this section in edgerc file that contains the credential set.')
+  // .option('--json [path]', 'Write command output to JSON file at given path, otherwise written to CLI cache directory')
+  .option('--accountkey <account-id>', 'internal parameter')
+  .on("option:debug", function () {
+    envUtils.setDebugMode(true);
+  })
+  .on("option:edgerc", function (edgeRcFilePath) {
+    envUtils.setEdgeRcFilePath(edgeRcFilePath);
+  })
+  .on("option:section", function (section) {
+    envUtils.setEdgeRcSection(section);
+  })
+  .on("option:json", function (path) {
+    edgeWorkersClientSvc.setJSONOutputMode(false); // set this to true when we enable json output mode
+    edgeWorkersClientSvc.setJSONOutputPath(path);
+  })
+  .on("option:accountkey", function (key) {
+    httpEdge.setAccountKey(key);
+  })
+  // this fires only when a command is not listed below with a custom action
+  .on('command:*', function (command) {
+    const firstCommand = command[0];
+    if (!this.commands.find(c => c._name == firstCommand)) {
+      cliUtils.logAndExit(1, `Invalid command: ${program.args.join(' ')}\nSee --help for a list of available commands.`);
+    }
+  })
+  .on("--help", function () {
+    cliUtils.logAndExit(0, copywrite);
+  });
+
+program
+  .command('help [command]')
+  .description('Displays help information for the given command.')
+  .action(function (arg) {
+    if (!arg) {
+      program.outputHelp();
+      cliUtils.logAndExit(0, copywrite);
+    }
+    else {
+      var command = (!!program.commands.find(c => c._name == arg)) ? program.commands.find(c => c._name == arg) : program.commands.find(c => c._aliases[0] == arg);
+      if (!command) {
+        cliUtils.logAndExit(1, `ERROR: Could not find a command for ${arg}`);
+      }
+      else {
+        command.outputHelp();
+      }
+    }
+  })
+  .on("--help", function () {
+    cliUtils.logAndExit(0, copywrite);
+  });
+
+
+program
+  .command("initialize")
+  .description("Initialize edgeKV for the first time")
+  .alias("init")
+  .action(async function () {
+    try {
+      await kvCliHandler.initializeEdgeKv();
+    } catch (e) {
+      cliUtils.logAndExit(1, e);
+    }
+  })
+  .on("--help", function () {
+    cliUtils.logAndExit(0, copywrite);
+  });
+
+  program.command("write <itemType> <environment> <namespace> <groupId> <itemId> <items>")
+  .description("Write an item to an EdgeKV Namespace")
+  .action(async function (itemType, environment, namespace, groupId, itemId, items) {
+    try {
+      await kvCliHandler.writeItemToEdgeKV(environment, namespace, groupId, itemId, items, itemType);
+    } catch (e) {
+      cliUtils.logAndExit(1, e);
+    }
+  })
+  .on("--help", function () {
+    cliUtils.logAndExit(0, copywrite);
+  });
+
+/* ========== The following are created as sub commands since commander does not support space  ========== */
```

Comment:
It's not apparent why these program commands are different?  What 'space' is referred to in the comment?

Reply:
literal " ", we want command with "create ns". With commander we will be able to do that only when we create as sub commands.

Reply:
I'm not sure it's best practice for a CLI command to include a space.  I wouldn't recommend this unless there is a very specific reason for it as it may cause confusion for users.

Reply:
This was a request from edgekv team. @mryakan can you please provide your thoughts on this ?